### PR TITLE
Include Associations

### PR DESCRIPTION
**Pull request by markolson**
Including associations fails the XML Schema validation test. Setting it as an empty array passes, as does including a single association object instead of an array.

The golden JSON test could be fixed as well, but I'd prefer to not change that until if we know the data is correct or not.

### CODE REVIEW

File: src/server/enrollment-system.js [hunk 1]
```diff
@@ -1062,8 +1086,17 @@ function veteranToFinancialsInfo(veteran) {
...
+function veteranToAssociationCollection(veteran) {
+  const associations = [];
+  const children = _.compact(veteran.children.map((child) => { return childToAssociation(child); }));
+  const spouse = spouseToAssociation(veteran);
+  associations.concat(children);
+  if (spouse) { associations.push(spouse); }
+  if (_.isEmpty(associations)) {
+    return undefined;
+  }
+  const result = associations.length === 1 ? associations.pop() : associations;
```
Comment (robbiethegeek) [inline src/server/enrollment-system.js]: @akainic like my use of a ternary :)

File: src/server/enrollment-system.js [hunk 2]
```diff
@@ -1094,13 +1092,11 @@ function veteranToAssociationCollection(veteran) {
...
   if (spouse) { associations.push(spouse); }
-
   if (_.isEmpty(associations)) {
     return undefined;
   }
-  const result = associations.map((association) => { return { association }; });
-  console.log(result);
-  return result;
+  const result = associations.length === 1 ? associations.pop() : associations;
+  return { association: result };
```
Comment (markolson) [inline src/server/enrollment-system.js]: I think the SOAP library will handle this correctly if it's left as a single item array, but I'll write some tests around it to verify

Comment (markolson) [inline src/server/enrollment-system.js]: @robbiethegeek after updating the conformance test XML, I changed this to just include `associations` as-is, and the results came out the same for `no-children.xml` which only has one `association` record.

Comment (robbiethegeek) [inline src/server/enrollment-system.js]: 👍 cool wasn't sure if we needed to manually do it or the soap library would thanks for checking.

**Comment by robbiethegeek** [comment]
LGTM @akainic want to give it a quick once over?

**Comment by akainic** [comment]
LGTM! Let's merge and test in staging and verify with Josh.

**Pull request merged**